spring-cloud-kubernetes-client-config/src/main/java/org/springframework/cloud/kubernetes/client/config/KubernetesClientConfigMapPropertySource.java

```diff
@@ -46,17 +47,18 @@ public KubernetesClientConfigMapPropertySource(CoreV1Api coreV1Api, String name,
 			Environment environment) {
 
 		try {
-			List<String> names = new ArrayList<>();
+			Set<String> names = new HashSet<>();
 			names.add(name);
 			if (environment != null) {
 				for (String activeProfile : environment.getActiveProfiles()) {
 					names.add(name + "-" + activeProfile);
 				}
 			}
-			Map<String, Object> result = new LinkedHashMap<>();
+			Map<String, Object> result = new HashMap<>();
 			coreV1Api.listNamespacedConfigMap(namespace, null, null, null, null, null, null, null, null, null, null)
 					.getItems().stream().filter(cm -> names.contains(cm.getMetadata().getName()))
-					.forEach(map -> result.putAll(processAllEntries(map.getData(), environment)));
```

Comment:
`AnyCollection::forEach` is different then `Stream::forEach`; so first collect to a `List`, then to a `forEach`